components/director/cmd/director/main.go

```diff
@@ -135,6 +140,26 @@ func main() {
 	}
 }
 
+func getPairingAdaptersMapping(filePath string) (map[string]string, error) {
+	if filePath == "" {
+		log.Infof("No configuration for pairing adapters")
+		return nil, nil
+	}
+	file, err := os.Open(filePath)
```

Comment:
In this approach, you need to close the file. Maybe you can use `ioutil.ReadFile` instead.